sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasPermissions.cs

```diff
@@ -3,56 +3,94 @@
 
 using System;
 using System.ComponentModel;
+using System.Runtime.CompilerServices;
 using System.Text;
+using Azure.Storage.Sas;
 
 namespace Azure.Storage.Sas
 {
     /// <summary>
     /// <see cref="BlobSasPermissions"/> supports reading and writing
-    /// permissions string for a blob's access policy.  Use <see cref="ToString"/>
+    /// permissions string for a blob's access policy.  Use ToString
     /// to generate a permissions string you can provide to
     /// <see cref="BlobSasBuilder.Permissions"/>.
     /// </summary>
-    public struct BlobSasPermissions : IEquatable<BlobSasPermissions>
+    [Flags]
+    public enum BlobSasPermissions
     {
+        /// <summary>
+        ///
+        /// </summary>
+        None = 0,
+
         /// <summary>
         /// Get or sets whether Read is permitted.
         /// </summary>
-        public bool Read { get; set; }
+        Read = 1,
 
         /// <summary>
         /// Get or sets whether Add is permitted.
         /// </summary>
-        public bool Add { get; set; }
+        Add = 2,
 
         /// <summary>
         /// Get or sets whether Create is permitted.
         /// </summary>
-        public bool Create { get; set; }
+        Create = 4,
 
         /// <summary>
         /// Get or sets whether Write is permitted.
         /// </summary>
-        public bool Write { get; set; }
+        Write = 8,
 
         /// <summary>
         /// Get or sets whether Delete is permitted.
         /// </summary>
-        public bool Delete { get; set; }
+        Delete = 16,
+
+        /// <summary>
+        ///
+        /// </summary>
+        All = ~None
+    }
+}
+
+namespace Azure.Storage.Blobs
+{
+    /// <summary>
+    /// Blob enum extensions
+    /// </summary>
+    internal static partial class BlobExtensions
+    {
 
         /// <summary>
         /// Create a permissions string to provide
         /// <see cref="BlobSasBuilder.Permissions"/>.
         /// </summary>
         /// <returns>A permissions string.</returns>
-        public override string ToString()
+        public static string ToPermissionsString(this BlobSasPermissions permissions)
         {
             var sb = new StringBuilder();
-            if (Read) { sb.Append(Constants.Sas.Permissions.Read); }
-            if (Add) { sb.Append(Constants.Sas.Permissions.Add); }
-            if (Create) { sb.Append(Constants.Sas.Permissions.Create); }
-            if (Write) { sb.Append(Constants.Sas.Permissions.Write); }
-            if (Delete) { sb.Append(Constants.Sas.Permissions.Delete); }
+            if ((permissions & BlobSasPermissions.Read) == BlobSasPermissions.Read)
+            {
+                sb.Append(Constants.Sas.Permissions.Read);
+            }
+            if ((permissions & BlobSasPermissions.Add) == BlobSasPermissions.Add)
+            {
+                sb.Append(Constants.Sas.Permissions.Add);
```

Comment:
Shouldn't you also be appending a delimiter, or do these string constants already have them? (They're not in the diff.)

Reply:
Nope, the string does not contain delimiters - each constant is just a single character.